pyfstat/helper_functions.py

```diff
@@ -1105,3 +1105,79 @@ def parse_list_of_numbers(val):
             f" Got exception: {e}."
         )
     return out
+
+
+def generate_loudest_file(
+    max_params,
+    tref,
+    outdir,
+    label,
+    sftfilepattern,
+    minStartTime=None,
+    maxStartTime=None,
+    transientWindowType=None,
+    earth_ephem=None,
+    sun_ephem=None,
+):
+    """Use lalapps_ComputeFstatistic_v2 to produce a .loudest file.
+
+    Parameters
+    -------
+    max_params: dict
+        Dictionary of a single parameter-space point.
+        This needs to already have been translated to lal conventions
+        and must NOT include detection statistic entries!
+    tref: int
+        Reference time of the max_params.
+    outdir: str
+        Directory to place the .loudest file in.
+    label: str
+        Search name bit to be used in the output filename.
+    sftfilepattern: str
+        Pattern to match SFTs using wildcards (`*?`) and ranges [0-9];
+        mutiple patterns can be given separated by colons.
+    minStartTime, maxStartTime: int or None
+        GPS seconds of the start time and end time;
+        default: use al available data.
+    transientWindowType: str or None
+        optional: transient window type,
+        needs to go with t0 and tau parameters inside max_params.
+    earth_ephem: str or None
+        optional: user-set Earth ephemeris file
+    sun_ephem: str or None
+        optional: user-set Sun ephemeris file
+
+    Returns
+    -------
+    loudest_file: str
+        The filename of the CFSv2 output file.
+    """
+    logging.info("Running CFSv2 to get .loudest file")
+    if np.any([key in max_params for key in ["delta_F0", "delta_F1", "tglitch"]]):
+        raise RuntimeError("CFSv2 --outputLoudest cannot deal with glitch parameters.")
+    if transientWindowType:
+        logging.warning(
+            "CFSv2 --outputLoudest always reports the maximum of the"
+            " standard CW 2F-statistic, not the transient max2F."
+        )
+
+    loudest_file = os.path.join(outdir, label + ".loudest")
+    cmd = "lalapps_ComputeFstatistic_v2 "
+    CFSv2_params = {
+        "DataFiles": f'"{sftfilepattern}"',
+        "outputLoudest": loudest_file,
+        "refTime": tref,
+    }
+    CFSv2_params.update(max_params)
+    opt_params = {
+        "minStartTime": minStartTime,
+        "maxStartTime": maxStartTime,
+        "transient-WindowType": transientWindowType,
+        "ephemEarth": earth_ephem,
+        "ephemSun": sun_ephem,
+    }
+    CFSv2_params.update({par: key for par, key in opt_params.items() if key})
```

Comment:
```suggestion
    CFSv2_params.update({par: val for par, val in opt_params.items() if val})
```

Just bugged by using key as a value

Reply:
Good catch, but let me make it even more canonical:
```suggestion
    CFSv2_params.update({key: val for key, val in opt_params.items() if val})
```